### ISSUE TITLE
`IsPublishable` property is not respected

### ISSUE THREAD

**Issue by augustfengd** [dotnet/sdk-container-builds#464]
Hello!

I've got a solution **Foo** that encompasses two projects, **Bar** and **Baz**, where **Baz** does not have the `Microsoft.NET.Build.Containers` package installed and is configured with `<IsPublishable>false</IsPublishable>`.

`dotnet publish --os linux --arch x64 /t:PublishContainer -c Release` will still try to attempt to build **Baz** as a container image.

```
dotnet publish --os linux --arch x64 /t:PublishContainer -c Release
MSBuild version 17.5.0+6f08c67f3 for .NET
  Determining projects to restore...
  Restored /Users/august.feng/learn/fsharp.48/Baz/Baz.fsproj (in 51 ms).
  Restored /Users/august.feng/learn/fsharp.48/Bar/Bar.fsproj (in 79 ms).
  Bar -> /Users/august.feng/learn/fsharp.48/Bar/bin/Release/net7.0/linux-x64/Bar.dll
  Bar -> /Users/august.feng/learn/fsharp.48/Bar/bin/Release/net7.0/linux-x64/publish/
/Users/august.feng/learn/fsharp.48/Baz/Baz.fsproj : error MSB4057: The target "PublishContainer" does not exist in the project.
  Building image 'bar' with tags 1.0.0 on top of base image mcr.microsoft.com:443/dotnet/runtime:7.0
  Pushed container 'bar:1.0.0' to local daemon
  ```

**Comment by baronfel** [comment]
Can you send a binlog or a sample project so I can see more details about the interaction here?

**Comment by augustfengd** [comment]
Hey @baronfel, I've just uploaded a sample project [here](https://github.com/augustfengd/sdk-container-builds-464). As you'll see in the sample, `Bar` is a console project.

**Comment by baronfel** [comment]
Awesome, thank you!

**Comment by baronfel** [comment]
Well the good news is that I wasn't able to repro this on .NET 8 previews. I'll try with the 7.0 series next.

**Comment by baronfel** [comment]
I also couldn't reproduce with 7.0.305 - what directory are you invoking the `publish` command from? `Bar`'s project folder? Or a level up at the solution level?

**Comment by augustfengd** [comment]
Hey, I'm running from the directory at the solution level. Also just confirming the environment, I'm on a Macbook Pro M1 with version 7.0.305 (arm64) and I'm getting the error.

**Comment by baronfel** [comment]
Ok, this makes more sense to me. When you specify targets at a solution level, those targets are invoked for all projects regardless of if those projects make sense to build that target. Because we don't support PublishProfiles for console apps (yet!) you are forced to specify the target to build.

The fastest way to fix this would be to stop calling `publish` from the solution directory, and instead publish just the application project that you need. This is my preferred option, as publish from a solution directory has very unclear semantics. You specified an OS and architecture as well - if you had any projects that didn't support that you'd get errors when doing this at the solution level. We really don't encourage doing that :)

Longer term, once we make PublishProfiles work for console apps then this stops being an issue. You can call `dotnet publish -p PublishProfile=DefaultContainer` and things will just work - the app that doesn't have the package/toggle the SDK-base publishing support won't load the PublishProfile and you'll be all set.

Medium term, we could add an additional Condition to the target, but I think it should be `Condition="'$(EnableSdkContainerSupport)' == 'true' and '$(IsPublishable)' == true'", not just the IsPublishable check.

**Comment by augustfengd** [comment]
Ah! Thanks for your insight on what's to come. We've got a large number of projects that are containerized so we're thinking of having a separate .sln file just for those projects as an immediate solution for now.

Both medium and long-term solution sounds good though.

**Comment by rainersigwald** [comment]
> we're thinking of having a separate .sln file just for those projects as an immediate solution for now.

Consider a [solution filter](https://devblogs.microsoft.com/visualstudio/load-only-the-projects-you-need-with-solution-filters) (they're easier to maintain than a whole separate solution).

**Comment by augustfengd** [comment]
@rainersigwald, big thanks for the solution filter suggestion. It's working out great. :)

**Comment by baronfel** [comment]
One big downside to adding this constraint is that is blows up publishing for console applications. All users of this tech that made console apps now have to set `IsPublishable` to `true` or else their containerization doesn't work anymore.

**Comment by baronfel** [comment]
This one's been added and will be shipped in 8.0.100 RC2!
